custom_components/bmw_connected_drive/sensor.py

```diff
@@ -41,12 +52,55 @@
     "remaining_range_fuel": ["mdi:map-marker-distance", LENGTH_MILES],
     "max_range_electric": ["mdi:map-marker-distance", LENGTH_MILES],
     "remaining_fuel": ["mdi:gas-station", VOLUME_GALLONS],
+    # LastTrip attributes
+    "avgCombinedConsumption": ["mdi:flash", f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_MILES}"],
+    "avgElectricConsumption": ["mdi:power-plug-outline", f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_MILES}"],
+    "avgRecuperation": ["mdi:recycle-variant", f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_MILES}"],
+    "electricDistance": ["mdi:map-marker-distance", LENGTH_MILES],
+    "savedFuel": ["mdi:fuel", VOLUME_GALLONS],
+    "totalDistance": ["mdi:map-marker-distance", LENGTH_MILES],
+    # AllTrips attributes
+    "chargecycleRange": ["mdi:map-marker-distance", LENGTH_MILES],
+    "totalElectricDistance": ["mdi:map-marker-distance", LENGTH_MILES],
+    "totalSavedFuel": ["mdi:fuel", VOLUME_GALLONS],
+}
+
+ATTR_TO_HA = {
     "charging_time_remaining": ["mdi:update", TIME_HOURS],
     "charging_status": ["mdi:battery-charging", None],
     # No icon as this is dealt with directly as a special case in icon()
     "charging_level_hv": [None, PERCENTAGE],
+    # LastTrip attributes
+    "date": ["mdi:calendar-blank", None],
+    "duration": ["mdi:timer-outline", TIME_MINUTES],
+    "electricDistanceRatio": ["mdi:percent-outline", PERCENTAGE],
+    # AllTrips attributes
+    "batterySizeMax": ["mdi:battery-charging-high", ENERGY_WATT_HOUR],
+    "resetDate": ["mdi:calendar-blank", None],
+    "savedCO2": ["mdi:tree-outline", MASS_KILOGRAMS],
+    "savedCO2greenEnergy": ["mdi:tree-outline", MASS_KILOGRAMS],
+    # ChargingProfile attributes
+    "climatizationEnabled": ["mdi:snowflake", None],
+    "preferredChargingWindow": ["mdi:dock-window", None],
+    "timer1": ["mdi:av-timer", None],
+    "timer2": ["mdi:av-timer", None],
+    "timer3": ["mdi:av-timer", None],
+    "overrideTimer": ["mdi:av-timer", None],
+    # Destination attributes
+    "Destination_0": ["mdi:pin-outline", None],
```

Comment:
Is 10 the max?
And does BMW also start with Destination_0? Destination_1 makes more sense to me.

Reply:
Changed it to allow limitless destinations